test/system/src/test/java/io/pravega/test/system/ConsumptionBasedRetentionWithMultipleReaderGroupsTest.java

```diff
@@ -283,4 +286,140 @@ public void multipleSubscriberCBRTest() throws Exception {
+        // Subscriber lower bound is 0/90, truncation should happen at this point
+        // The timeout is set to 3 minutes a little longer than the retention period which is set to 2 minutes
+        // in order to confirm that the retention has taken place.
+        AssertExtensions.assertEventuallyEquals("Truncation did not take place at offset 90.", true, () -> controller.getSegmentsAtTime(
+                        new StreamImpl(SCOPE_1, STREAM_1), 0L).join().values().stream().anyMatch(off -> off == 90),
+                5000, 3 * 60 * 1000L);
+
+        ReaderGroupConfig nonSubscriberReaderGroupConfig = ReaderGroupConfig
+                .builder()
+                .retentionType(ReaderGroupConfig.StreamDataRetention.NONE)
+                .disableAutomaticCheckpoints().stream(Stream.of(SCOPE_1, STREAM_1)).build();
+        //Changing the readergroup from subscriber to non-subscriber
+        readerGroup.resetReaderGroup(nonSubscriberReaderGroupConfig);
+        ReaderGroupConfig.StreamDataRetention readerGroupRetentionType = controller.getReaderGroupConfig(SCOPE_1, READER_GROUP_3).join().getRetentionType();
+        assertEquals(ReaderGroupConfig.StreamDataRetention.NONE, readerGroupRetentionType);
+        assertEquals(0, controller.listSubscribers(SCOPE_1, STREAM_1).join().size());
+
+        // Fill 5 more events to the stream.
+        for (int i = 1; i <= 5; i++) {
+            log.info("Writing event to {}/{}", SCOPE_1, STREAM_1);
+            writer.writeEvent(SIZE_30_EVENT).join();
+        }
+
+        // Retention set has two stream cut at 0/210...0/360
+        // READER_GROUP_3 is not a subscriber as its retention type is updated to NONE
+        // The timeout is set to 3 minutes a little longer than the retention period which is set to 2 minutes
+        // in order to confirm that the retention has taken place.
+        // Check to make sure truncation happened at streamcut in the retention set
+        AssertExtensions.assertEventuallyEquals("Truncation did not take place at offset 210.", true, () -> controller.getSegmentsAtTime(
+                        new StreamImpl(SCOPE_1, STREAM_1), 0L).join().values().stream().anyMatch(off -> off == 210),
+                5000, 3 * 60 * 1000L);
+
+        //Changing the readergroup from non-subscriber to subscriber again
+        readerGroup.resetReaderGroup(readerGroupConfig);
+        readerGroupRetentionType = controller.getReaderGroupConfig(SCOPE_1, READER_GROUP_3).join().getRetentionType();
+        assertEquals(ReaderGroupConfig.StreamDataRetention.MANUAL_RELEASE_AT_USER_STREAMCUT, readerGroupRetentionType);
+        assertEquals(1, controller.listSubscribers(SCOPE_1, STREAM_1).join().size());
+
+        // Recreates the reader
+        reader = clientFactory.createReader(READER_GROUP_3 + "-" + 1, READER_GROUP_3, new JavaSerializer<>(),
+                ReaderConfig.builder().build());
+
+        // fill stream with 3 events
+        for (int i = 1; i <= 3; i++) {
```

Comment:
Also add the scenario to verify what happens when stream has more data than MAX_SIZE_IN_STREAM by adding 8 events instead of 3. // Retention set has two stream cut at 0/360, 0/600(instead of 450), then read only 3 events and generate streamcut so SLB is 0/300.

Reply:
After adding 3 events , stream has more data than MAX_SIZE_IN_STREAM (ie, 450-210 =240 which is greater than 180 max size). Do we still need to modify here ?